lib/less/rails/railtie.rb

```diff
@@ -14,11 +14,16 @@ class Railtie < ::Rails::Railtie
         require 'less-rails'
         Sprockets::Engines #force autoloading
 
-        if Sprockets.respond_to?(:register_engine)
-          args = ['.less', LessTemplate]
-          args << { mime_type: 'text/less', silence_deprecation: true } if Sprockets::VERSION.start_with?("3")
-          env.register_engine(*args)
-        end
+        config.assets.configure do |env|
+          if env.respond_to?(:register_engine)
+            args = ['.less', LessTemplate]
+            args << { mime_type: 'text/less', silence_deprecation: true } if env::VERSION.start_with?("3")
+            env.register_engine(*args)
+          elsif env.respond_to?(:register_transformer)
```

Comment:
This was only a test i've submitted to @simi  just to let him check if
working against rails 5.x... (It also fails validation against travis-ci
because i've not implemented testingbat all!!) The mod was written in less
than 30' just to workaround the first deprecation i've noted in issue
#122... Sure it May be refined with a Little effort in time!!! Feel free to
suggestion modifications or any other way to male it work better!!

Many thanks for you time,
Regards

Francesco

Il 08/set/2016 06:52 PM, "Maxim Dobryakov" <notifications@github.com> ha
scritto:

> In lib/less/rails/railtie.rb
> <https://github.com/metaskills/less-rails/pull/124#discussion_r78045287>:
>
> > @@ -14,11 +14,16 @@ class Railtie < ::Rails::Railtie
> >          require 'less-rails'
> >          Sprockets::Engines #force autoloading
> >
> > -        if Sprockets.respond_to?(:register_engine)
> > -          args = ['.less', LessTemplate]
> > -          args << { mime_type: 'text/less', silence_deprecation: true } if Sprockets::VERSION.start_with?("3")
> > -          env.register_engine(*args)
> > -        end
> > +        config.assets.configure do |env|
> > +          if env.respond_to?(:register_engine)
> > +            args = ['.less', LessTemplate]
> > +            args << { mime_type: 'text/less', silence_deprecation: true } if env::VERSION.start_with?("3")
> > +            env.register_engine(*args)
> > +          elsif env.respond_to?(:register_transformer)
>
> Very strange to check register_transformer method and don't use it. Will
> be better to find version when this deprecation warning was added and use
> comparison with this version instead.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/metaskills/less-rails/pull/124/files/9745707b4795d5be262956a23a38454145a71988#r78045287>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AG_whcyi4X-JouE6iuh_ktycwPVtUQ6_ks5qoD1cgaJpZM4Jw-Pm>
> .
>